crowbar_framework/lib/crowbar/upgrade.rb

```diff
@@ -100,6 +100,17 @@ def knife_files
           filecontent_replace(file, "nova_dashboard", "horizon")
         end
       end
+
+      # find admin node and update target_platform
+      nodes_path = knife_path.join("nodes")
+      nodes_path.children.each do |file|
+        json = JSON.load(file.read)
+        next unless json["crowbar"] && json["crowbar"]["admin_node"]
+        json["target_platform"] = "suse-12.1"
```

Comment:
Btw, we should probably unset `node[:provisioner][:default_os]` for the admin node, to have that be updated too (although it seems it will be updated anyway right now, but that could be a bug).